x/mint/abci.go

```diff
@@ -43,3 +48,20 @@ func BeginBlocker(ctx sdk.Context, k Keeper) {
 		),
 	)
 }
+
+// getNewAverageBlockTime calculates the average time it takes to forge a new block based on the cumulated
+// moving average since the beginning of the chain.
+// On first block we use params.BlocksPerYear to calculate it, so the average in block one is:
+// NanosecondsInAYear / params.BlocksPerYear
+func getNewAverageBlockTime(ctx sdk.Context, params types.Params, minter types.Minter) time.Duration {
+	if minter.LastBlockTimestamp.IsZero() { // Comes from Genesis
+		nanoSecondsABlock := types.YEAR.Nanoseconds() / int64(params.BlocksPerYear)
```

Comment:
idk...It seems silly to keep around the `BlocksPerYear` param just for this single case which will only ever be executed once. I would try to think of a cleaner approach. I can think of a few:

1. Subtract current time from genesis time
2. Rename `BlocksPerYear` to `InitialBlockTime`
3. Use Tendermint's pre-commit timeout